### ISSUE TITLE
[Proposal] Aws ssm config provider

### ISSUE THREAD

**Issue by klagupl** [open-telemetry/opentelemetry-collector#6137]
**Is your feature request related to a problem? Please describe.**
We have many lambdas running on aws that use the opentelemtry extension. With this solution we could centralize storage of the configs. Also it would be more secure.

**Describe the solution you'd like**
Create  a new config provider that will retrieve the otel collector config from aws parameter store.
The schema would be like this: ssm:parameter_name

**Additional context**
With this solution, we can centralize storing all otel configs.

**Comment by bogdandrutu** [comment]
I agree with the request, though I don't want to have all providers in core, let's think how we can have this in contrib and available as part of our contrib distribution?

**Comment by klagupl** [comment]
You're right, it should be in contrib. 
Should i move this request to contrib repo then?
